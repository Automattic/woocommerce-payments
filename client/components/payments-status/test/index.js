/** @format */
/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import PaymentsStatus from '../';

describe( 'PaymentsStatus', () => {
	test( 'renders enabled status', () => {
		const { container: paymentsStatus } = renderPaymentsStatus( 1, 20 );
		expect( paymentsStatus ).toMatchSnapshot();
	} );

	test( 'renders disabled status', () => {
		const { container: paymentsStatus } = renderPaymentsStatus( 0, 20 );
		expect( paymentsStatus ).toMatchSnapshot();
	} );

	function renderPaymentsStatus( paymentsEnabled, iconSize ) {
		return render(
			<PaymentsStatus
				paymentsEnabled={ paymentsEnabled }
				iconSize={ iconSize }
			/>
		);
	}
} );
