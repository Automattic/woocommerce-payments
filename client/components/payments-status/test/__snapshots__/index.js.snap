// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentsStatus renders disabled status 1`] = `
<div>
  <span
    class="account-status__info__red"
  >
    <svg
      class="gridicon gridicons-notice"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 15h-2v-2h2v2zm0-4h-2l-.5-6h3l-.5 6z"
        />
      </g>
    </svg>
    Disabled
  </span>
</div>
`;

exports[`PaymentsStatus renders enabled status 1`] = `
<div>
  <span
    class="account-status__info__green"
  >
    <svg
      class="gridicon gridicons-checkmark-circle"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M11 17.768l-4.884-4.884 1.768-1.768L11 14.232l8.658-8.658C17.823 3.39 15.075 2 12 2 6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10c0-1.528-.353-2.97-.966-4.266L11 17.768z"
        />
      </g>
    </svg>
    Enabled
  </span>
</div>
`;

exports[`PaymentsStatus renders pending verification status 1`] = `
<div>
  <span
    class="account-status__info__gray"
  >
    <svg
      class="gridicon gridicons-notice"
      height="20"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 15h-2v-2h2v2zm0-4h-2l-.5-6h3l-.5 6z"
        />
      </g>
    </svg>
    Pending verification
  </span>
</div>
`;
